packages/headless/src/api/commerce/product-listings/product-listing-params.ts

```diff
@@ -34,6 +38,47 @@ export interface ProductListingRequestParam {
   userContext?: ContextPayload;
 }
 
+// TODO: Use interfaces for nested params
+export interface ProductListingV2RequestParam {
+  listingId: string;
+  locale: string;
+  mode: string;
+  clientId: string;
+  selectedFacets?: {
+    facetId?: string;
+    type?: string;
+    field?: string;
+    values?: {
+      value?: string;
+      start?: number | string;
+      end?: number | string;
+      isInclusive?: boolean;
+    }[];
+  }[];
+  selectedPage?: {
+    page?: number;
+  };
+  selectedSort?: SortCriterion;
+  context: {
+    user: {
+      userId?: string;
+      email?: string;
+      ipAddress: string;
+      userAgent: string;
```

Comment:
I mean we could probably do without them

I believe the user agent and IP address can both be extracted from the request headers on the API side

And if the user is logged in, then we should also be able to retrieve their email address from the access token... Unless API key authentication is used in the app, in which case passing the email would be legit.

Reply:
I've asked your question on the spec: https://docs.google.com/spreadsheets/d/1-WFpnipA3fgwCX3GhvnmJvPSwO_jfiQY-lZytsapfKs/edit?disco=AAAA3Gak_T0

As for the email, API key authentication will be used. This means we can't extract the email from the api key.

Reply:
Yeah I understand that API key authentication would be legit in a typical commerce use case since every user would likely see the same results.

However, I think the API should have a mechanism to extract the email from the search token if present, and fall back on the email request parameter otherwise

If some sort of authentication is already implemented in a solution, it's not a huge extra step for the implementer to add search token authentication as well. And I suppose there are also cases like B2B where not all users would see the same results?